src/main/java/org/broadinstitute/hellbender/tools/exome/sexgenotyper/PloidyAnnotatedTargetCollection.java

```diff
@@ -0,0 +1,219 @@
+        /* assert the lists and non-empty */
+        if (targetList.size() == 0) {
+            throw new UserException.BadInput("Target list can not be empty");
+        }
+        if (contigAnnotsList.size() == 0) {
+            throw new UserException.BadInput("Contig ploidy annotation list can not be empty");
+        }
+
+        /* assert targets have unique names */
+        final Map<String, Long> targetNameCounts = targetList.stream()
+                .map(Target::getName)
+                .collect(Collectors.groupingBy(Function.identity(), Collectors.counting()));
+        if (targetNameCounts.keySet().size() < targetList.size()) {
+            throw new UserException.BadInput("Targets must have unique names. Non-unique target names: " +
+                    targetNameCounts.keySet().stream()
+                            .filter(name -> targetNameCounts.get(name) > 1)
+                            .collect(Collectors.joining(", ")));
+        }
+
+        /* assert annotated contigs are unique */
+        final Map<String, Long> contigAnnotsCounts = contigAnnotsList.stream()
+                .map(ContigPloidyAnnotation::getContigName)
+                .collect(Collectors.groupingBy(Function.identity(), Collectors.counting()));
+        if (contigAnnotsCounts.keySet().size() < contigAnnotsList.size()) {
+            throw new UserException.BadInput("Some contigs are multiply annotated: " +
+                    contigAnnotsCounts.keySet().stream()
+                            .filter(contig -> contigAnnotsCounts.get(contig) > 1) /* multiply annotated contigs */
+                            .collect(Collectors.joining(", ")));
+        }
+
+        /* assert all contigs present in the target list are annotated */
+        final Set<String> contigNamesFromTargets = targetList.stream()
+                .map(Target::getContig).collect(Collectors.toSet());
+        final Set<String> contigNamesFromAnnots = contigAnnotsList.stream()
+                .map(ContigPloidyAnnotation::getContigName).collect(Collectors.toSet());
+        final Set<String> missingContigs = Sets.difference(contigNamesFromTargets, contigNamesFromAnnots);
+        if (missingContigs.size() > 0) {
+            throw new UserException.BadInput("All contigs must be annotated. Annotations are missing for: " +
+                    missingContigs.stream().collect(Collectors.joining(", ")));
+        }
+
+        /* assert all contigs have annotations for all ploidy classes */
+        final Set<String> firstAnnotPloidyTagSet = contigAnnotsList.get(0).getGenotypesSet();
+        if (contigAnnotsList.stream().filter(annot -> !annot.getGenotypesSet().equals(firstAnnotPloidyTagSet)).count() > 0) {
+            throw new UserException.BadInput("Not all entries in the contig ploidy annotation list have the same " +
+                    "ploidy tag set");
+        }
+    }
+
+    @Override
+    public int targetCount() {
```

Comment:
The ones in the superclass suffices. If you want to _add_ something to the superclass methods, you use `@inheritDoc` followed by additional stuff. `@inheritDoc` prepends the javadoc from the superclass javadoc.